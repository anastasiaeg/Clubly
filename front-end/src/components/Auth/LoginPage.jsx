import React, { Component } from "react";
import {
  Button,
  Card,
  ControlGroup,
  Elevation,
  H2,
  InputGroup,
  Text
} from "@blueprintjs/core";
import { Link } from "react-router-dom";
import { Redirect } from "react-router-dom";
import "./login.css";
import * as routes from "../../constants/routes";
import { authService } from "../../services/authService";

class LoginPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      height: window.innerHeight,
      width: window.innerWidth,
      loggedIn: false
    };
  }

  userInfo = {
    email: "",
    password: ""
  };

  updateDimensions = () => {
    this.setState({
      height: window.innerHeight,
      width: window.innerWidth
    });
  };

  componentWillMount() {
    this.updateDimensions();
  }

  componentDidMount() {
    window.addEventListener("resize", this.updateDimensions);
    this.updateDimensions();
  }

  componentWillUnmount() {
    window.removeEventListener("resize", this.updateDimensions);
  }

  onLogin = event => {
    if (typeof event.preventDefault !== "undefined") {
      event.preventDefault();
    }
    authService
      .login(this.userInfo.email, this.userInfo.password)
      .then(response => {
        if (response.data) {
          this.setState({
            loggedIn: true
          });
        }
      });
    return false;
  };

  render() {
    const { width, height, loggedIn } = this.state;
    if (loggedIn) {
      return (
        <Redirect
          to={{
            pathname:
              this.userInfo.email === "admin@ncsu.edu" ? routes.ADMIN : "/",
            state: { from: this.props.location }
          }}
        />
      );
    }
    return (
      <div className="login" style={{ width: width, height: height }}>
        <Card elevation={Elevation.TWO} className="login-card">
          <H2>Welcome to Clubly!</H2>
          <br />
          <form onSubmit={this.onLogin}>
            <ControlGroup vertical={true}>
              <InputGroup
                large={true}
                placeholder="Email"
                type="email"
                onChange={element => {
                  this.userInfo.email = element.target.value;
                }}
              />
              <InputGroup
                large={true}
                placeholder="Password"
                type="password"
                onChange={element => {
                  this.userInfo.password = element.target.value;
                }}
              />
              <Button
                large={true}
                icon="arrow-right"
                text="Login"
                type="submit"
              />
            </ControlGroup>
          </form>
          <br />
          <Text>
            <Link to={routes.SIGN_UP}>Don't have an account?</Link>
          </Text>
        </Card>
      </div>
    );
  }
}

export default LoginPage;
