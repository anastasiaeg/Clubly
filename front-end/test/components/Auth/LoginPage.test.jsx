import React from "react";

import LoginPage from "../../../src/components/Auth/LoginPage";
// test file
import { shallow } from "enzyme";

describe("Login Page", () => {
  test("should render", () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
    wrapper.setState({ loggedIn: true });
    const form = wrapper.find("form").at(0);
  });
});
