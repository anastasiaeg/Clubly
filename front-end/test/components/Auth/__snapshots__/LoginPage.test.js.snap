// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Page should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Blueprint3.Card
          className="login-card"
          elevation={2}
          interactive={false}
>
          <Unknown>
                    Welcome to Clubly!
          </Unknown>
          <br />
          <form
                    onSubmit={[Function]}
          >
                    <Blueprint3.ControlGroup
                              vertical={true}
                    >
                              <Blueprint3.InputGroup
                                        large={true}
                                        onChange={[Function]}
                                        placeholder="Email"
                                        type="email"
                              />
                              <Blueprint3.InputGroup
                                        large={true}
                                        onChange={[Function]}
                                        placeholder="Password"
                                        type="password"
                              />
                              <Blueprint3.Button
                                        icon="arrow-right"
                                        large={true}
                                        text="Login"
                                        type="submit"
                              />
                    </Blueprint3.ControlGroup>
          </form>
          <br />
          <Blueprint3.Text>
                    <a>
                              Don't have an account?
                    </a>
          </Blueprint3.Text>
</Blueprint3.Card>,
        null,
      ],
      "className": "login",
      "style": Object {
        "height": 768,
        "width": 1024,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Unknown>
              Welcome to Clubly!
</Unknown>,
            <br />,
            <form
              onSubmit={[Function]}
>
              <Blueprint3.ControlGroup
                            vertical={true}
              >
                            <Blueprint3.InputGroup
                                          large={true}
                                          onChange={[Function]}
                                          placeholder="Email"
                                          type="email"
                            />
                            <Blueprint3.InputGroup
                                          large={true}
                                          onChange={[Function]}
                                          placeholder="Password"
                                          type="password"
                            />
                            <Blueprint3.Button
                                          icon="arrow-right"
                                          large={true}
                                          text="Login"
                                          type="submit"
                            />
              </Blueprint3.ControlGroup>
</form>,
            <br />,
            <Blueprint3.Text>
              <a>
                            Don't have an account?
              </a>
</Blueprint3.Text>,
          ],
          "className": "login-card",
          "elevation": 2,
          "interactive": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Welcome to Clubly!",
            },
            "ref": null,
            "rendered": "Welcome to Clubly!",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Blueprint3.ControlGroup
                vertical={true}
>
                <Blueprint3.InputGroup
                                large={true}
                                onChange={[Function]}
                                placeholder="Email"
                                type="email"
                />
                <Blueprint3.InputGroup
                                large={true}
                                onChange={[Function]}
                                placeholder="Password"
                                type="password"
                />
                <Blueprint3.Button
                                icon="arrow-right"
                                large={true}
                                text="Login"
                                type="submit"
                />
</Blueprint3.ControlGroup>,
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Blueprint3.InputGroup
                    large={true}
                    onChange={[Function]}
                    placeholder="Email"
                    type="email"
/>,
                  <Blueprint3.InputGroup
                    large={true}
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
/>,
                  <Blueprint3.Button
                    icon="arrow-right"
                    large={true}
                    text="Login"
                    type="submit"
/>,
                ],
                "vertical": true,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "large": true,
                    "onChange": [Function],
                    "placeholder": "Email",
                    "type": "email",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "large": true,
                    "onChange": [Function],
                    "placeholder": "Password",
                    "type": "password",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "icon": "arrow-right",
                    "large": true,
                    "text": "Login",
                    "type": "submit",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": "form",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <a>
                Don't have an account?
</a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Don't have an account?",
              },
              "ref": null,
              "rendered": "Don't have an account?",
              "type": "a",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      null,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Blueprint3.Card
            className="login-card"
            elevation={2}
            interactive={false}
>
            <Unknown>
                        Welcome to Clubly!
            </Unknown>
            <br />
            <form
                        onSubmit={[Function]}
            >
                        <Blueprint3.ControlGroup
                                    vertical={true}
                        >
                                    <Blueprint3.InputGroup
                                                large={true}
                                                onChange={[Function]}
                                                placeholder="Email"
                                                type="email"
                                    />
                                    <Blueprint3.InputGroup
                                                large={true}
                                                onChange={[Function]}
                                                placeholder="Password"
                                                type="password"
                                    />
                                    <Blueprint3.Button
                                                icon="arrow-right"
                                                large={true}
                                                text="Login"
                                                type="submit"
                                    />
                        </Blueprint3.ControlGroup>
            </form>
            <br />
            <Blueprint3.Text>
                        <a>
                                    Don't have an account?
                        </a>
            </Blueprint3.Text>
</Blueprint3.Card>,
          null,
        ],
        "className": "login",
        "style": Object {
          "height": 768,
          "width": 1024,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Unknown>
                Welcome to Clubly!
</Unknown>,
              <br />,
              <form
                onSubmit={[Function]}
>
                <Blueprint3.ControlGroup
                                vertical={true}
                >
                                <Blueprint3.InputGroup
                                                large={true}
                                                onChange={[Function]}
                                                placeholder="Email"
                                                type="email"
                                />
                                <Blueprint3.InputGroup
                                                large={true}
                                                onChange={[Function]}
                                                placeholder="Password"
                                                type="password"
                                />
                                <Blueprint3.Button
                                                icon="arrow-right"
                                                large={true}
                                                text="Login"
                                                type="submit"
                                />
                </Blueprint3.ControlGroup>
</form>,
              <br />,
              <Blueprint3.Text>
                <a>
                                Don't have an account?
                </a>
</Blueprint3.Text>,
            ],
            "className": "login-card",
            "elevation": 2,
            "interactive": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Welcome to Clubly!",
              },
              "ref": null,
              "rendered": "Welcome to Clubly!",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Blueprint3.ControlGroup
                  vertical={true}
>
                  <Blueprint3.InputGroup
                                    large={true}
                                    onChange={[Function]}
                                    placeholder="Email"
                                    type="email"
                  />
                  <Blueprint3.InputGroup
                                    large={true}
                                    onChange={[Function]}
                                    placeholder="Password"
                                    type="password"
                  />
                  <Blueprint3.Button
                                    icon="arrow-right"
                                    large={true}
                                    text="Login"
                                    type="submit"
                  />
</Blueprint3.ControlGroup>,
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": Array [
                    <Blueprint3.InputGroup
                      large={true}
                      onChange={[Function]}
                      placeholder="Email"
                      type="email"
/>,
                    <Blueprint3.InputGroup
                      large={true}
                      onChange={[Function]}
                      placeholder="Password"
                      type="password"
/>,
                    <Blueprint3.Button
                      icon="arrow-right"
                      large={true}
                      text="Login"
                      type="submit"
/>,
                  ],
                  "vertical": true,
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "large": true,
                      "onChange": [Function],
                      "placeholder": "Email",
                      "type": "email",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "large": true,
                      "onChange": [Function],
                      "placeholder": "Password",
                      "type": "password",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "icon": "arrow-right",
                      "large": true,
                      "text": "Login",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": "form",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <a>
                  Don't have an account?
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Don't have an account?",
                },
                "ref": null,
                "rendered": "Don't have an account?",
                "type": "a",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        null,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
